ReactCommon/hermes/executor/HermesExecutorFactory.cpp

```diff
@@ -164,6 +164,7 @@ class DecoratedRuntime : public jsi::WithRuntimeDecorator<ReentrancyCheck> {
         runtime_, hermesRuntime_, jsQueue);
     facebook::hermes::inspector::chrome::enableDebugging(
         std::move(adapter), "Hermes React Native");
+    hermesRuntime_.registerForProfiling();
```

Comment:
This forces all RN users to register for profiling, regardless of their choice. The more "correct" way of doing this would be to set the [HermesConfig::EnableSampleProfiling](https://github.com/facebook/hermes/blob/main/public/hermes/Public/RuntimeConfig.h#L87) in the runtime config used to create the JS executor.